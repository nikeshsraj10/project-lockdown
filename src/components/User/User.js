import React, {Component} from 'react';
import axios from 'axios';
import classes from './User.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';

import COUNTRIES from '../../constants/countries';
import DatePicker from 'react-datepicker';
import "react-datepicker/dist/react-datepicker.css";

import 'font-awesome/css/font-awesome.min.css';

const initState = {
    firstName: '',
    lastName: '',
    email: '',
    gender: 1,
    birthdate: null,
    country: COUNTRIES[0],
    mobile: '',
    portraitURL: '',
    errorObj: {
        firstName: true,
        email: true,
        birthdate: true,
        mobile: true,
        portraitURL: true,
        formValid: true
    }

}

export default class User extends Component{
    constructor(props){
        super(props);
        this.state = initState;
    }

    handlePersonalDetailsChange = (event) => {
        let {name, value} = event.target;
        if(name === 'gender')
            value = +value;
        this.setState({
            [name]: value
        })
    }

    handleBirthdayChange = (date) => {
        this.setState({
            birthdate: date
        });
    }

    handleCountryChange = (event) => {
        const {name, value} = event.target;
        this.setState({
            [name]: value
        })
    }

    validateForm = () => {
        let errorObj = {
            firstName: false,
            email: false,
            birthdate: false,
            mobile: false,
            portraitURL: true, //true because its not mandatory but if entered, it should be valid
            formValid: false
        }
        //firstName validation
        if(!this.state.firstName){
            errorObj.firstName = false;
        }else{
            //Check for Valid name i.e. alphanumeric only
            let regex = new RegExp(/^[a-z0-9]+$/, 'i');
            if(regex.test(this.state.firstName))
                errorObj.firstName = true
            else
                errorObj.firstName = false;
        }
        //firstName validation
        if(!this.state.email){
            errorObj.email = false;
        }else{
            //Check for Valid email
            let regex = new RegExp(/^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/);
            if(regex.test(this.state.email))
                errorObj.email = true
            else
                errorObj.email = false;
        }
        //Validate Birthdate
        if(!this.state.birthdate)
            errorObj.birthdate = false;
        else
            errorObj.birthdate = true;
        //Validate Mobile Number
        if(!this.state.mobile){
            errorObj.mobile = false;
        }else{
            //Check for Mobile Number i.e. start with + or numbers
            let regex = new RegExp(/^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/, 'im');
            if(regex.test(this.state.mobile))
                errorObj.mobile = true
            else
                errorObj.mobile = false;
        }
        if(this.state.portraitURL){
            let regex = new RegExp(/\w+\.(jpg|jpeg|gif|png|tiff|bmp)$/, 'gi');
            if(regex.test(this.state.portraitURL))
                errorObj.portraitURL = true
            else
                errorObj.portraitURL = false;
        }
        if(errorObj.firstName && errorObj.mobile && errorObj.birthdate && errorObj.email && errorObj.portraitURL){
            //Form is Valid
            errorObj.formValid = true;
            return errorObj;
        }else{
            errorObj.formValid = false;
            return errorObj;
        }
    }

    handleSubmit = (event) => {
        event.preventDefault();
        const errorObj = this.validateForm();
        if(errorObj.formValid){
            const user = {
                firstName: this.state.firstName,
                lastName: this.state.lastName,
                email: this.state.email,
                gender: this.state.gender,
                birthdate: this.state.birthdate,
                country: this.state.country,
                mobile: this.state.mobile,
                portraitURL: this.state.portraitURL
            }
            axios.post('http://localhost:5000/users/add', user)
                 .then(res => {
                     console.log(res.data); 
                 })
                 .catch(err => console.log(err));
            this.setState({
                ...initState
            })
            window.location = '/'
        }else{
            this.setState({
                errorObj: {...errorObj}
            })
            return;
        }
    }

    render(){
        return(
            <React.Fragment>
                <div className={classes.page}>
                    {/* Form Header */}
                    <div >
                        <h2 className={["progress-bar", "bg-warning", "text-muted", "pl-2", classes.heading].join(" ")}>Creating a new User</h2>
                    </div>
                    {/* Form Sections Overview */}
                    <div className="col-sm-9">
                        {/* <div><i className="fa fa-spinner fa-spin">Font Awesome</i></div> */}
                        <div className={[classes['inline-block'], classes.icons].join(' ')}>
                            <i className="fas fa-user fa-3x"></i>
                            <span className={classes.left}>1.Personal details</span>    
                        </div>
                        <div className={[classes['inline-block'], classes.icons].join(' ')}>
                            <i className="fas fa-star fa-3x"></i>
                            <span className={classes.left}>2. Your favorites</span>
                        </div>
                    </div>
                    {/* Form Begins here */}
                    <div className="container">
                        <div className="text-center pl-4">
                            <h2 className={classes['align-left']}>Your Personal Details</h2>
                        </div>
                        <div className="col-sm-12">
                            <form name="personalDetailsForm" className="needs-validation" >
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                        <label htmlFor="firstName" >First Name*: </label>
                                        <input  type="text" className="form-control" id="firstName" name="firstName"
                                                placeholder="First Name" value={this.state.firstName} maxLength="20"
                                                onChange={this.handlePersonalDetailsChange} required/>
                                    </div>
                                    {!this.state.errorObj.firstName ? <div className="pl-3" style={{color: 'red'}}>
                                    Please enter a valid First Name
                                    </div> : null}
                                </div>
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                        <label htmlFor="lastName" >Last Name: </label>
                                        <input  type="text" className="form-control" id="lastName" name="lastName"
                                                placeholder="Last Name" value={this.state.lastName} maxLength="20"
                                                onChange={this.handlePersonalDetailsChange}/>
                                </div>
    
                                </div>
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                        <label htmlFor="email" >Email Address*: </label>
                                        <input  type="email" className="form-control" id="email" name="email"
                                                placeholder="you@example.com" value={this.state.email}
                                                onChange={this.handlePersonalDetailsChange} required={true} />
                                    </div>
                                    {!this.state.errorObj.email ? <div className="pl-3" style={{color: 'red'}}>
                                        Please enter a valid email address
                                    </div> : null}
                                </div>
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                    <label htmlFor="gender">Gender*: </label>
                                    <div className="form-group">
                                        <span className={["custom-control", "custom-checkbox", "p-3", classes.left].join(' ')}>
                                            <label  htmlFor="male">
                                             <input name="gender" type="radio" value={1}
                                                    className="form-check-input" checked={this.state.gender === 1}
                                                    onChange={this.handlePersonalDetailsChange} required />Male
                                            </label>
                                            
                                        </span>
                                        <span className={["custom-control", "custom-checkbox", "p-3", classes.left].join(' ')}>
                                            <label  htmlFor="female">
                                                <input  name="gender" type="radio" value={0}
                                                        className="form-check-input" checked={this.state.gender === 0}
                                                        onChange={this.handlePersonalDetailsChange} required />Female
                                            </label>
                                            
                                        </span>
                                        <span className={["custom-control", "custom-checkbox", "p-3", classes.left].join(' ')}>
                                            <label  htmlFor="unspecified">
                                                 <input name="gender" type="radio" value={-1}
                                                        className="form-check-input" checked={this.state.gender === -1}
                                                        onChange={this.handlePersonalDetailsChange} required />Unspecified
                                            </label>
                                            
                                        </span>
                                    </div>
                                    </div>
                                </div>
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                        <label htmlFor="portraitURL" >Portrait URL*: </label>
                                        <input  type="text" className="form-control" id="portraitURL" name="portraitURL"
                                                placeholder="Share your picture" value={this.state.portraitURL} required
                                                onChange={this.handlePersonalDetailsChange} />
                                    </div>
                                    {!this.state.errorObj.portraitURL ? <div className="pl-3" style={{color: 'red'}}>
                                        Please enter a valid URL
                                    </div> : null }
                                </div>
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                        <label htmlFor="birthday" >Date of Birth*: </label>
                                        <div className="input-group-addon col-sm-12" style={{textAlign: "initial"}}>
                                            <DatePicker selected={this.state.birthdate ? this.state.birthdate : null} 
                                                        onChange={this.handleBirthdayChange} />
                                        </div>
                                    </div>
                                    {!this.state.errorObj.birthdate ? <div className="pl-3" style={{color: 'red'}}>
                                        Please select your Date of Birth
                                    </div> : null }
                                </div>
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                        <label htmlFor="country" >Country*: </label>
                                        <select name="country" onChange={this.handleCountryChange} value={this.state.country} className="custom-select d-block w-100" id="country">
                                            {
                                                COUNTRIES.map((country, index) => <option key={index}>{country}</option>)
                                            }
                                        </select>
                                    </div>
                                </div>
                                <div className="row">
                                    <div className="col-sm-12 mb-3">
                                        <label htmlFor="mobile" >Mobile Number*: </label>
                                        <input  type="text" className="form-control" id="mobile" name="mobile"
                                                placeholder="Mobile" value={this.state.mobile} required
                                                onChange={this.handlePersonalDetailsChange} />
                                    </div>
                                    {!this.state.errorObj.mobile ? <div className="pl-3" style={{color: 'red'}}>
                                        Please enter a valid mobile number
                                    </div> : null}
                                </div>
                                <hr className="mb-4"/>
                                <button className="btn btn-primary btn-lg btn-block" onClick={this.handleSubmit} type="submit">Add User</button>
                            </form>
                        </div>
                    </div>
                </div>
            </React.Fragment>
        );
    }
}