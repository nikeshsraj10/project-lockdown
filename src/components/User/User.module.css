.page{
    padding: 2% 2%;
    margin: 1%;
    box-sizing: border-box;
}

.heading{
    text-align: initial;
}

.inline-block{
    display: inline-block;
}

.icons{
    position: relative;
    max-width: 100px;
    z-index: 2;
}

.left{
    float: left;
}

.react-datepicker__input-container input {
    width: 100% !important;
}

.react-datepicker-ignore-onclickoutside {
    width: 100%;
}

.align-left{
    text-align: left;
}