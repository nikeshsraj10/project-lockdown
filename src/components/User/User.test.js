import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import User from './User';

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<User></User>, div);
});

it("Add User to be in the document", () => {
    const { getByTestId } = render(<User/>)
    expect(getByTestId('addUserButton')).toBeInTheDocument();
})


it("Update User button should not be in the document", () => {
    const { queryByTestId } = render(<User/>)
    expect(queryByTestId('updateUser')).toBeNull();
})

it("matches snapshot", () => {
    const tree = renderer.create(<User></User>).toJSON();
    expect(tree).toMatchSnapshot();
}) 