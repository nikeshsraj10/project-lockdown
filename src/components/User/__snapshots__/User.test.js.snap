// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="page"
>
  <div>
    <h2
      className="progress-bar bg-warning text-muted pl-2 heading"
    >
      Creating a new User
    </h2>
  </div>
  <div
    className="container"
  >
    <div
      className="text-center pl-4"
    >
      <h2
        className="align-left"
      >
        Your Personal Details
      </h2>
    </div>
    <div
      className="col-sm-12"
    >
      <form
        className="needs-validation"
        name="personalDetailsForm"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="firstName"
            >
              First Name*: 
            </label>
            <input
              className="form-control"
              id="firstName"
              maxLength="20"
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              readOnly={false}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="lastName"
            >
              Last Name: 
            </label>
            <input
              className="form-control"
              id="lastName"
              maxLength="20"
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              readOnly={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="email"
            >
              Email Address*: 
            </label>
            <input
              className="form-control"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="you@example.com"
              readOnly={false}
              required={true}
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="gender"
            >
              Gender*: 
            </label>
            <div
              className="form-group"
            >
              <span
                className="custom-control custom-checkbox p-3 left"
              >
                <label
                  htmlFor="male"
                >
                  <input
                    checked={true}
                    className="form-check-input"
                    disabled={false}
                    name="gender"
                    onChange={[Function]}
                    required={true}
                    type="radio"
                    value={1}
                  />
                  Male
                </label>
              </span>
              <span
                className="custom-control custom-checkbox p-3 left"
              >
                <label
                  htmlFor="female"
                >
                  <input
                    checked={false}
                    className="form-check-input"
                    disabled={false}
                    name="gender"
                    onChange={[Function]}
                    required={true}
                    type="radio"
                    value={0}
                  />
                  Female
                </label>
              </span>
              <span
                className="custom-control custom-checkbox p-3 left"
              >
                <label
                  htmlFor="unspecified"
                >
                  <input
                    checked={false}
                    className="form-check-input"
                    disabled={false}
                    name="gender"
                    onChange={[Function]}
                    required={true}
                    type="radio"
                    value={-1}
                  />
                  Unspecified
                </label>
              </span>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="portraitURL"
            >
              Portrait URL: 
            </label>
            <input
              className="form-control"
              id="portraitURL"
              name="portraitURL"
              onChange={[Function]}
              placeholder="Share your picture"
              readOnly={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="birthday"
            >
              Date of Birth*: 
            </label>
            <div
              className="input-group-addon col-sm-12"
              style={
                Object {
                  "textAlign": "initial",
                }
              }
            >
              <div
                className="react-datepicker-wrapper"
              >
                <div
                  className="react-datepicker__input-container"
                >
                  <input
                    className=""
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    readOnly={false}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="country"
            >
              Country*: 
            </label>
            <select
              className="custom-select d-block w-100"
              disabled={false}
              id="country"
              name="country"
              onChange={[Function]}
              value="Afghanistan"
            >
              <option>
                Afghanistan
              </option>
              <option>
                Ã…land Islands
              </option>
              <option>
                Albania
              </option>
              <option>
                Algeria
              </option>
              <option>
                American Samoa
              </option>
              <option>
                Andorra
              </option>
              <option>
                Angola
              </option>
              <option>
                Anguilla
              </option>
              <option>
                Antarctica
              </option>
              <option>
                Antigua and Barbuda
              </option>
              <option>
                Argentina
              </option>
              <option>
                Armenia
              </option>
              <option>
                Aruba
              </option>
              <option>
                Australia
              </option>
              <option>
                Austria
              </option>
              <option>
                Azerbaijan
              </option>
              <option>
                Bahamas
              </option>
              <option>
                Bahrain
              </option>
              <option>
                Bangladesh
              </option>
              <option>
                Barbados
              </option>
              <option>
                Belarus
              </option>
              <option>
                Belgium
              </option>
              <option>
                Belize
              </option>
              <option>
                Benin
              </option>
              <option>
                Bermuda
              </option>
              <option>
                Bhutan
              </option>
              <option>
                Bolivia
              </option>
              <option>
                Bosnia and Herzegovina
              </option>
              <option>
                Botswana
              </option>
              <option>
                Bouvet Island
              </option>
              <option>
                Brazil
              </option>
              <option>
                British Indian Ocean Territory
              </option>
              <option>
                Brunei Darussalam
              </option>
              <option>
                Bulgaria
              </option>
              <option>
                Burkina Faso
              </option>
              <option>
                Burundi
              </option>
              <option>
                Cambodia
              </option>
              <option>
                Cameroon
              </option>
              <option>
                Canada
              </option>
              <option>
                Cape Verde
              </option>
              <option>
                Cayman Islands
              </option>
              <option>
                Central African Republic
              </option>
              <option>
                Chad
              </option>
              <option>
                Chile
              </option>
              <option>
                China
              </option>
              <option>
                Christmas Island
              </option>
              <option>
                Cocos (Keeling) Islands
              </option>
              <option>
                Colombia
              </option>
              <option>
                Comoros
              </option>
              <option>
                Congo
              </option>
              <option>
                Congo, The Democratic Republic of The
              </option>
              <option>
                Cook Islands
              </option>
              <option>
                Costa Rica
              </option>
              <option>
                Cote D'ivoire
              </option>
              <option>
                Croatia
              </option>
              <option>
                Cuba
              </option>
              <option>
                Cyprus
              </option>
              <option>
                Czech Republic
              </option>
              <option>
                Denmark
              </option>
              <option>
                Djibouti
              </option>
              <option>
                Dominica
              </option>
              <option>
                Dominican Republic
              </option>
              <option>
                Ecuador
              </option>
              <option>
                Egypt
              </option>
              <option>
                El Salvador
              </option>
              <option>
                Equatorial Guinea
              </option>
              <option>
                Eritrea
              </option>
              <option>
                Estonia
              </option>
              <option>
                Ethiopia
              </option>
              <option>
                Falkland Islands (Malvinas)
              </option>
              <option>
                Faroe Islands
              </option>
              <option>
                Fiji
              </option>
              <option>
                Finland
              </option>
              <option>
                France
              </option>
              <option>
                French Guiana
              </option>
              <option>
                French Polynesia
              </option>
              <option>
                French Southern Territories
              </option>
              <option>
                Gabon
              </option>
              <option>
                Gambia
              </option>
              <option>
                Georgia
              </option>
              <option>
                Germany
              </option>
              <option>
                Ghana
              </option>
              <option>
                Gibraltar
              </option>
              <option>
                Greece
              </option>
              <option>
                Greenland
              </option>
              <option>
                Grenada
              </option>
              <option>
                Guadeloupe
              </option>
              <option>
                Guam
              </option>
              <option>
                Guatemala
              </option>
              <option>
                Guernsey
              </option>
              <option>
                Guinea
              </option>
              <option>
                Guinea-bissau
              </option>
              <option>
                Guyana
              </option>
              <option>
                Haiti
              </option>
              <option>
                Heard Island and Mcdonald Islands
              </option>
              <option>
                Holy See (Vatican City State)
              </option>
              <option>
                Honduras
              </option>
              <option>
                Hong Kong
              </option>
              <option>
                Hungary
              </option>
              <option>
                Iceland
              </option>
              <option>
                India
              </option>
              <option>
                Indonesia
              </option>
              <option>
                Iran, Islamic Republic of
              </option>
              <option>
                Iraq
              </option>
              <option>
                Ireland
              </option>
              <option>
                Isle of Man
              </option>
              <option>
                Israel
              </option>
              <option>
                Italy
              </option>
              <option>
                Jamaica
              </option>
              <option>
                Japan
              </option>
              <option>
                Jersey
              </option>
              <option>
                Jordan
              </option>
              <option>
                Kazakhstan
              </option>
              <option>
                Kenya
              </option>
              <option>
                Kiribati
              </option>
              <option>
                Korea, Democratic People's Republic of
              </option>
              <option>
                Korea, Republic of
              </option>
              <option>
                Kuwait
              </option>
              <option>
                Kyrgyzstan
              </option>
              <option>
                Lao People's Democratic Republic
              </option>
              <option>
                Latvia
              </option>
              <option>
                Lebanon
              </option>
              <option>
                Lesotho
              </option>
              <option>
                Liberia
              </option>
              <option>
                Libyan Arab Jamahiriya
              </option>
              <option>
                Liechtenstein
              </option>
              <option>
                Lithuania
              </option>
              <option>
                Luxembourg
              </option>
              <option>
                Macao
              </option>
              <option>
                Macedonia, The Former Yugoslav Republic of
              </option>
              <option>
                Madagascar
              </option>
              <option>
                Malawi
              </option>
              <option>
                Malaysia
              </option>
              <option>
                Maldives
              </option>
              <option>
                Mali
              </option>
              <option>
                Malta
              </option>
              <option>
                Marshall Islands
              </option>
              <option>
                Martinique
              </option>
              <option>
                Mauritania
              </option>
              <option>
                Mauritius
              </option>
              <option>
                Mayotte
              </option>
              <option>
                Mexico
              </option>
              <option>
                Micronesia, Federated States of
              </option>
              <option>
                Moldova, Republic of
              </option>
              <option>
                Monaco
              </option>
              <option>
                Mongolia
              </option>
              <option>
                Montenegro
              </option>
              <option>
                Montserrat
              </option>
              <option>
                Morocco
              </option>
              <option>
                Mozambique
              </option>
              <option>
                Myanmar
              </option>
              <option>
                Namibia
              </option>
              <option>
                Nauru
              </option>
              <option>
                Nepal
              </option>
              <option>
                Netherlands
              </option>
              <option>
                Netherlands Antilles
              </option>
              <option>
                New Caledonia
              </option>
              <option>
                New Zealand
              </option>
              <option>
                Nicaragua
              </option>
              <option>
                Niger
              </option>
              <option>
                Nigeria
              </option>
              <option>
                Niue
              </option>
              <option>
                Norfolk Island
              </option>
              <option>
                Northern Mariana Islands
              </option>
              <option>
                Norway
              </option>
              <option>
                Oman
              </option>
              <option>
                Pakistan
              </option>
              <option>
                Palau
              </option>
              <option>
                Palestinian Territory, Occupied
              </option>
              <option>
                Panama
              </option>
              <option>
                Papua New Guinea
              </option>
              <option>
                Paraguay
              </option>
              <option>
                Peru
              </option>
              <option>
                Philippines
              </option>
              <option>
                Pitcairn
              </option>
              <option>
                Poland
              </option>
              <option>
                Portugal
              </option>
              <option>
                Puerto Rico
              </option>
              <option>
                Qatar
              </option>
              <option>
                Reunion
              </option>
              <option>
                Romania
              </option>
              <option>
                Russian Federation
              </option>
              <option>
                Rwanda
              </option>
              <option>
                Saint Helena
              </option>
              <option>
                Saint Kitts and Nevis
              </option>
              <option>
                Saint Lucia
              </option>
              <option>
                Saint Pierre and Miquelon
              </option>
              <option>
                Saint Vincent and The Grenadines
              </option>
              <option>
                Samoa
              </option>
              <option>
                San Marino
              </option>
              <option>
                Sao Tome and Principe
              </option>
              <option>
                Saudi Arabia
              </option>
              <option>
                Senegal
              </option>
              <option>
                Serbia
              </option>
              <option>
                Seychelles
              </option>
              <option>
                Sierra Leone
              </option>
              <option>
                Singapore
              </option>
              <option>
                Slovakia
              </option>
              <option>
                Slovenia
              </option>
              <option>
                Solomon Islands
              </option>
              <option>
                Somalia
              </option>
              <option>
                South Africa
              </option>
              <option>
                South Georgia and The South Sandwich Islands
              </option>
              <option>
                Spain
              </option>
              <option>
                Sri Lanka
              </option>
              <option>
                Sudan
              </option>
              <option>
                Suriname
              </option>
              <option>
                Svalbard and Jan Mayen
              </option>
              <option>
                Swaziland
              </option>
              <option>
                Sweden
              </option>
              <option>
                Switzerland
              </option>
              <option>
                Syrian Arab Republic
              </option>
              <option>
                Taiwan (ROC)
              </option>
              <option>
                Tajikistan
              </option>
              <option>
                Tanzania, United Republic of
              </option>
              <option>
                Thailand
              </option>
              <option>
                Timor-leste
              </option>
              <option>
                Togo
              </option>
              <option>
                Tokelau
              </option>
              <option>
                Tonga
              </option>
              <option>
                Trinidad and Tobago
              </option>
              <option>
                Tunisia
              </option>
              <option>
                Turkey
              </option>
              <option>
                Turkmenistan
              </option>
              <option>
                Turks and Caicos Islands
              </option>
              <option>
                Tuvalu
              </option>
              <option>
                Uganda
              </option>
              <option>
                Ukraine
              </option>
              <option>
                United Arab Emirates
              </option>
              <option>
                United Kingdom
              </option>
              <option>
                United States
              </option>
              <option>
                United States Minor Outlying Islands
              </option>
              <option>
                Uruguay
              </option>
              <option>
                Uzbekistan
              </option>
              <option>
                Vanuatu
              </option>
              <option>
                Venezuela
              </option>
              <option>
                Viet Nam
              </option>
              <option>
                Virgin Islands, British
              </option>
              <option>
                Virgin Islands, U.S.
              </option>
              <option>
                Wallis and Futuna
              </option>
              <option>
                Western Sahara
              </option>
              <option>
                Yemen
              </option>
              <option>
                Zambia
              </option>
              <option>
                Zimbabwe
              </option>
            </select>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-12 mb-3"
          >
            <label
              htmlFor="mobile"
            >
              Mobile Number*: 
            </label>
            <input
              className="form-control"
              id="mobile"
              name="mobile"
              onChange={[Function]}
              placeholder="Mobile"
              readOnly={false}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <hr
          className="mb-4"
        />
        <button
          className="btn btn-primary btn-lg btn-block"
          data-testid="addUserButton"
          onClick={[Function]}
          type="submit"
        >
          Add User
        </button>
      </form>
    </div>
  </div>
</div>
`;
